### PR TITLE
Removed deprecation warning about backend.auth.keys

### PR DESCRIPTION
**Pull request by 04kash**
## Description of the change

Fixed default configuration to remove deprecation warning about backend.auth.keys by replacing it with https://github.com/backstage/backstage/blob/master/beps/0007-auth-external-services/README.md#the-legacy-access-type

## Existing or Associated Issue(s)

Fixes: https://issues.redhat.com/browse/RHIDP-2716

## Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the `values.yaml` and added to the README.md. The [pre-commit](https://pre-commit.com/) utility can be used to generate the necessary content. Use `pre-commit run -a` to apply changes.
- [x] JSON Schema template updated and re-generated the raw schema via `pre-commit` hook.
- [ ] List tests pass for Chart using the [Chart Testing](https://github.com/helm/chart-testing) tool and the `ct lint` command.

### CODE REVIEW

File: charts/backstage/values.yaml [hunk 1]
```diff
@@ -59,8 +59,11 @@ upstream:
             password: ${POSTGRESQL_ADMIN_PASSWORD}
             user: postgres
         auth:
-          keys:
-            - secret: ${BACKEND_SECRET}
+          externalAccess:
+            - type: legacy
+              options:
+                subject: legacy-default-cofig
```
Comment (coreydaley) [inline charts/backstage/values.yaml:65]: Is this a typo? Should it be `legacy-default-config`?

**Comment by coreydaley** [comment]
I think that you also need to run the pre-commit hook to regenerate the schema and update the README

File: charts/backstage/values.yaml (hunk 1 above)
Comment (04kash) [reply charts/backstage/values.yaml:65]: Oh yes, that's a typo

File: charts/backstage/values.yaml [hunk 2]
```diff
@@ -5,7 +5,7 @@ global:
     includes:
       # -- List of dynamic plugins included inside the `janus-idp/backstage-showcase` container image, some of which are disabled by default.
       # This file ONLY works with the `janus-idp/backstage-showcase` container image.
-      - 'dynamic-plugins.default.yaml'
+      - "dynamic-plugins.default.yaml"
```
Comment (coreydaley) [inline charts/backstage/values.yaml:8]: These should stay single quotes: https://support.asg.com/mob/mvw/10_0/mv_ag/using_quotes_with_yaml_special_characters.htm#:~:text=Single%20quotes%20let%20you%20put,returned%20as%20the%20string%20%5Cn.&text=Double%20quotes%20parse%20escape%20codes,as%20a%20line%20feed%20character.

File: charts/backstage/values.yaml [hunk 3]
```diff
@@ -111,7 +114,7 @@ upstream:
 
     args:
       # This additional `app-config`` file is generated by the initContainer below, and contains the merged configuration of installed dynamic plugins.
-      - '--config'
+      - "--config"
```
Comment (coreydaley) [inline charts/backstage/values.yaml:117]: These should stay single quotes: https://support.asg.com/mob/mvw/10_0/mv_ag/using_quotes_with_yaml_special_characters.htm#:~:text=Single%20quotes%20let%20you%20put,returned%20as%20the%20string%20%5Cn.&text=Double%20quotes%20parse%20escape%20codes,as%20a%20line%20feed%20character.

**Review by rm3l — APPROVED** [review]
Works fine - thanks @04kash !! For the reasons mentioned in https://issues.redhat.com/browse/RHIDP-2716?focusedId=25057410&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-25057410, this PR should not be cherry-picked to `1.2.x`.

**Review by rm3l — APPROVED** [review]
Tests not passing because the `latest` tag for [`quay.io/janus-idp/backstage-showcase`](https://quay.io/repository/janus-idp/backstage-showcase?tab=tags&tag=latest) is outdated, and the changes here won't work on RHDH 1.1

<details>
<summary>Logs</summary>

```
/opt/app-root/src/node_modules/@backstage/backend-app-api/dist/index.cjs.js:1481
          throw new Error(
                ^

Error: Failed to instantiate service 'core.tokenManager' for 'search' because the factory function threw an error, Error: Failed to instantiate service 'core.tokenManager' because createRootContext threw an error, Error: You must configure at least one key in backend.auth.keys for production.
    at /opt/app-root/src/node_modules/@backstage/backend-app-api/dist/index.cjs.js:1481:17
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async BackendInitializer.getInitDeps_fn (/opt/app-root/src/node_modules/@backstage/backend-app-api/dist/index.cjs.js:1644:20)
    at async /opt/app-root/src/node_modules/@backstage/backend-app-api/dist/index.cjs.js:1779:32
    at async processNode (/opt/app-root/src/node_modules/@backstage/backend-app-api/dist/index.cjs.js:1302:22)
    at async Promise.all (index 0)
    at async processMoreNodes (/opt/app-root/src/node_modules/@backstage/backend-app-api/dist/index.cjs.js:1297:7)
    at async _DependencyGraph.parallelTopologicalTraversal (/opt/app-root/src/node_modules/@backstage/backend-app-api/dist/index.cjs.js:1308:5)
    at async /opt/app-root/src/node_modules/@backstage/backend-app-api/dist/index.cjs.js:1777:9
    at async Promise.all (index 4)
    at async BackendInitializer.doStart_fn (/opt/app-root/src/node_modules/@backstage/backend-app-api/dist/index.cjs.js:1757:3)
    at async BackendInitializer.start (/opt/app-root/src/node_modules/@backstage/backend-app-api/dist/index.cjs.js:1611:5)
    at async BackstageBackend.start (/opt/app-root/src/node_modules/@backstage/backend-app-api/dist/index.cjs.js:1881:5)

Node.js v18.18.2
```

</details>

I think `quay.io/janus-idp/backstage-showcase:latest` should be updated to the latest stable version of the Showcase app.
See https://redhat-internal.slack.com/archives/C04CUSD4JSG/p1720191475569659

/hold

**Comment by coreydaley** [comment]
@rm3l So do we need to change the image here to point to next instead?

**Comment by rm3l** [comment]
> @rm3l So do we need to change the image here to point to next instead?

We discussed in https://redhat-internal.slack.com/archives/C04CUSD4JSG/p1720191475569659 (https://issues.redhat.com/browse/RHIDP-3157) that it would probably make more sense to use the downstream RHDH image here.
@04kash Could you please update the image in `values.yaml` to use `quay.io/rhdh/rhdh-hub-rhel9:latest` instead of the outdated `quay.io/janus-idp/backstage-showcase:latest` ? The test should pass with the RHDH image..
I'm making similar change in the operator repo.

**Comment by 04kash** [comment]
> @04kash Are you ready for us to merge this?

Yes

**Pull request merged**